Sources/SwifterAuth.swift

```diff
@@ -116,9 +116,21 @@ public extension Swifter {
     }
     #endif
     
-    public class func handleOpenURL(_ url: URL) {
+    public class func handleOpenURL(_ url: URL, callbackURL: URL) -> Bool {
+        guard isTwitterURLScheme(url, callbackURL: callbackURL) else {
+            return false
+        }
         let notification = Notification(name: .swifterCallback, object: nil, userInfo: [CallbackNotification.optionsURLKey: url])
         NotificationCenter.default.post(notification)
+        return true;
+    }
+    
+    public class func isTwitterURLScheme(_ url: URL, callbackURL: URL) -> Bool {
```

Comment:
Since this doesn't really target Twitter URL scheme specifically, it can be moved to an extension under `URL` like such:

```swift
public extension URL {
  func hasSameUrlScheme(as otherUrl: URL) -> Bool {
    guard let scheme = self.scheme, let otherScheme = otherUrl.scheme else { return false }
    return scheme.caseInsensitiveCompare(otherScheme) == .orderedSame
  }
}
```

Reply:
I moved the url logic code to URL++.swift file